tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3326,6 +3345,51 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Adds a [[PreTransBinding]] in front a result [[PreTransform]].
+   *
+   *  This can force the binding if the result is a [[PreTransGenTree]].
+   */
+  private def addPreTransBinding(preTransBinding: PreTransBinding,
+      result: PreTransform): PreTransform = {
+    result match {
+      case result: PreTransBlock =>
+        PreTransBlock(preTransBinding, result)
+      case result: PreTransLocalDef =>
+        PreTransBlock(preTransBinding, result)
+      case PreTransRecordTree(tree, tpe, cancelFun) =>
+        PreTransRecordTree(
+            finishTransformBindings(Left(preTransBinding) :: Nil, tree),
+            tpe, cancelFun)
+      case PreTransTree(tree, tpe) =>
+        PreTransTree(
+            finishTransformBindings(Left(preTransBinding) :: Nil, tree),
+            tpe)
+    }
```

Comment:
Am I missing something, or why can't you just replace this body by: `addPreTransBinding(Left(preTransBinding) :: Nil, result)`?